src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -97,6 +97,55 @@ public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 cer
             return this;
         }
 
+
+        /// <summary>
+        /// Sets the certificate associated with the application
+        /// </summary>
+        /// <param name="certificate">The X509 certificate used as credentials to prove the identity of the application to Azure AD.</param>
+        /// <param name="withSendX5C"><c>true</c> if the x5c should be sent. Otherwise <c>false</c>.
+        /// The default is <c>false</c></param>
+        /// <remarks>You should use certificates with a private key size of at least 2048 bytes. Future versions of this library might reject certificates with smaller keys. </remarks>
+        public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 certificate, bool withSendX5C)
+        {
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            if (!certificate.HasPrivateKey)
+            {
+                throw new MsalClientException(MsalError.CertWithoutPrivateKey, MsalErrorMessage.CertMustHavePrivateKey(nameof(certificate)));
+            }
+
+            Config.ClientCredentialCertificate = certificate;
+            Config.ConfidentialClientCredentialCount++;
+            return WithSendX5C(withSendX5C);
+        }
+
+        /// <summary>
+        /// Specifies if the x5c claim (public key of the certificate) should be sent to the STS.
+        /// Sending the x5c enables application developers to achieve easy certificate roll-over in Azure AD:
+        /// this method will send the public certificate to Azure AD along with all STS requests,
+        /// so that Azure AD can use it to validate the subject name based on a trusted issuer policy.
+        /// This saves the application admin from the need to explicitly manage the certificate rollover
+        /// (either via portal or PowerShell/CLI operation). For details see https://aka.ms/msal-net-sni
+        /// </summary>
+        /// <param name="withSendX5C"><c>true</c> if the x5c should be sent. Otherwise <c>false</c>.
+        /// The default is <c>false</c></param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public ConfidentialClientApplicationBuilder WithSendX5C(bool withSendX5C)
+        {
```

Comment:
i don't think we need this one. It has to go with the cert or signed assertion, so makes sense that it's an additional parameter there, so no need to have it separate. I think it would lead to confusion and people setting it twice.

Reply:
Generally agreed. Thankfully setting it twice won't do anything bad. The only reason I went this way was to snap with how the other SDKs do it (e.g. [Java](https://www.javadoc.io/doc/com.microsoft.azure/msal4j/latest/com/microsoft/aad/msal4j/ConfidentialClientApplication.Builder.html)) - happy to remove.  Other consideration was that since SNI is still largely 1p only, I wasn't sure about polluting the public namespace so visibly with something they shouldn't use. WDYT - remove it?

Reply:
I would do one or the other, but not both. I guess Jean-Marc would need to decide. What happens if a 3rd party tries to use it? also it's false by default, so most won't even notice it.

Reply:
3ps that try to use this will see a perf impact, but won't break (it doesn't help or harm them beyond sending extraneous bits over the wire).  Classic AAD behaviour would be to ignore the x5c claim if the cert thumbprint exists already in the app/SP registration, because while the developer is doing something "wrong" they're not doing anything illegal, so we'll let them do it.

Reply:
It's kind of a non-optional parameter for first party, so i think it goes outside our normal "let's not pollute". 
@jmprieur to decide.

Reply:
Could be an option to change the default of sending the X5C to true

Reply:
It is server to server communication (i.e. high bandwidth). So it impact may not be as much. It is a trade of between reliability and perf. The perf impact will be smaller compared to the gain. - My 2 cents